Userland/Libraries/LibWeb/DOM/Document.cpp

```diff
@@ -2289,6 +2289,71 @@ void Document::set_cookie(StringView cookie_string, Cookie::Source source)
     page().client().page_did_set_cookie(m_url, cookie.value(), source);
 }
 
+String Document::fg_color() const
+{
+    if (auto* body_element = body())
```

Comment:
The specification says that:

> [...] When there is no body element or if it is a frameset element, the attributes must instead return the empty string on getting and do nothing on setting.

So I think this something like this is needed in each of your getters and setters:

```suggestion
    if (auto* body_element = body(); body_element && !is<HTML::HTMLFrameSetElement>(*body_element))
```

This accounts for the case where `body()` returns a `HTMLFrameSetElement`.

---

FYI, this page has a test case that covers the behavior that is expected if `body` is a frameset element: 

http://wpt.live/html/obsolete/requirements-for-implementations/other-elements-attributes-and-apis/document-color-01.html

All of the tests on this page should pass if these attributes are implemented correctly.